src/main/java/com/tjtanjin/steve/tasks/TaskHandler.java

```diff
@@ -0,0 +1,162 @@
+            if (task.getStatus().equals("complete")) {
+                return "Error: Task is already completed!";
+            } else {
+                task.markCompleted();
+                String taskName = task.getTaskName();
+                String taskType = task.getType().toUpperCase();
+                LocalDate[] taskDates = task.getDates();
+                if (STORAGE_HANDLER.saveTask(index, "DONE",
+                        taskName, "complete", taskType, taskDates)) {
+                    return "Info: Yay your task is done! :D";
+                } else {
+                    return "Error: There was an error marking your task as done, "
+                            + "please check your syntax and try again.";
+                }
+            }
+        } catch (IndexOutOfBoundsException e) {
+            return "Error: The specified task index does not exist!";
+        }
+    }
+
+    /**
+     * Deletes a task.
+     *
+     * @param index index of task to delete
+     * @return string response after operation is done
+     */
+    public String deleteTask(int index) {
+        try {
+            Task task = TASKS.get(index);
+            TASKS.remove(task);
+            String taskName = task.getTaskName();
+            String taskType = task.getType().toUpperCase();
+            LocalDate[] taskDates = task.getDates();
+            if (STORAGE_HANDLER.saveTask(index, "DELETE",
+                    taskName, "complete", taskType, taskDates)) {
+                return "Info: The following task has been deleted:\n" + task;
+            } else {
+                return "Error: There was an error deleting your task, "
+                        + "please check your syntax and try again.";
+            }
+        } catch (IndexOutOfBoundsException e) {
+            return "Error: The specified task index does not exist!";
+        }
+    }
+
+    /**
+     * Shows all tasks to user
+     *
+     * @return string response after operation is done
+     */
+    public String showAllTasks() {
+        if (TASKS.size() == 0) {
+            return "Info: You have no task at the moment!";
+        } else {
+            StringBuilder str = new StringBuilder("Info: ");
+            for (int i = 1; i <= TASKS.size(); i++) {
+                Task task = TASKS.get(i - 1);
+                str.append(i).append(".").append(task).append("\n");
+            }
+            str.append("You have ").append(TASKS.size()).append(" task(s)!");
+            return str.toString();
+        }
+    }
+
+    /**
+     * Finds all tasks whose name contains input from user.
+     *
+     * @param taskNames names of task to check against
+     * @return string response after operation is done
+     */
+    public String findTask(String ... taskNames) {
+        if (TASKS.size() == 0) {
+            return "Info: You have no task at the moment!";
+        } else {
+            int counter = 0;
+            StringBuilder str = new StringBuilder("Info: ");
```

Comment:
Perhaps rename str to something more descriptive like stringOfFoundTasks.

Reply:
Did not realise I used such an ambiguous variable name, thank you for highlighting this :D